### PR TITLE
Use Controller instead of ContainerAware

### PR DESCRIPTION
**Pull request by core23**

### CODE REVIEW

**Comment by Soullivaneuh** [comment]
Looks like goot to me. :+1: 

Ping @sonata-project/contributors for a second review.

File: Controller/AdminSecurityController.php [hunk 1]
```diff
@@ -12,29 +12,29 @@
 namespace Sonata\UserBundle\Controller;
 
 use FOS\UserBundle\Model\UserInterface;
-use Symfony\Component\DependencyInjection\ContainerAware;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
 use Symfony\Component\HttpFoundation\RedirectResponse;
```
Comment (OskarStark) [inline Controller/AdminSecurityController.php]: needed anymore?

**Comment by OskarStark** [comment]
Looks good to me!

File: Controller/AdminSecurityController.php (hunk 1 above)
Comment (core23) [inline Controller/AdminSecurityController.php]: What do you mean? The Controller is needed for this ^^

Comment (OskarStark) [inline Controller/AdminSecurityController.php]: I mean, that this use-statement seems unused...

Comment (Soullivaneuh) [inline Controller/AdminSecurityController.php]: It's used here: https://github.com/sonata-project/SonataUserBundle/pull/641/files#diff-b773cf909994196ea448ba8b717c1c55R24

BTW, if it's not use, php-cs-fixer would detect it. :+1:

Comment (OskarStark) [inline Controller/AdminSecurityController.php]: Yes, but only used in the PHP Doc :)

https://github.com/core23/SonataUserBundle/blob/patch-4/Controller/AdminSecurityController.php

Comment (Soullivaneuh) [inline Controller/AdminSecurityController.php]: Yeah, and next? It's not a problem. ;-)

Comment (dbu) [inline Controller/AdminSecurityController.php]: having use statement for phpdoc comments makes us see better what this class depends on, even if its about return values that have no type hint in php 5. and it helps IDEs to understand the code and autocomplete. +1 to keep it that way.

File: Controller/AdminSecurityController.php [hunk 2]
```diff
@@ -12,29 +12,29 @@
...
-            $url = $this->container->get('router')->generate('sonata_admin_dashboard');
+            $this->get('session')->getFlashBag()->set('sonata_user_error', 'sonata_user_already_authenticated');
+            $url = $this->generateUrl('sonata_admin_dashboard');
 
-            return new RedirectResponse($url);
+            return $this->redirect($url);
         }
 
-        $request = $this->container->get('request');
+        $request = $this->get('request');
```
Comment (Soullivaneuh) [inline Controller/AdminSecurityController.php]: Request should be passed on the method with null default value and a `resolveRequest` should be written for BC.

File: Controller/AdminSecurityController.php (hunk 1 above)
Comment (OskarStark) [inline Controller/AdminSecurityController.php]: no problem, just a info @Soullivaneuh 

@dbu thanks for the info, so i'm :+1: too to keep this line!

File: Controller/AdminSecurityController.php (hunk 2 above)
Comment (core23) [inline Controller/AdminSecurityController.php]: Fixed

**Comment by core23** [comment]
@OskarStark RTM

**Comment by OskarStark** [comment]
Thank you @core23

File: Controller/AdminSecurityController.php [hunk 3]
```diff
@@ -12,29 +12,32 @@
...
-class AdminSecurityController extends ContainerAware
+class AdminSecurityController extends Controller
 {
     /**
+     * @param Request $request
+     *
      * @return Response|RedirectResponse
      */
-    public function loginAction()
+    public function loginAction(Request $request = null)
```
Comment (Trip09) [inline Controller/AdminSecurityController.php]: Missing the import of Request

**Comment by Soullivaneuh** [comment]
@OskarStark please at least, wait for all CI returns before PHP code modification merging.

@core23 can you please quickfix @Trip09 note? Thanks.

**Comment by core23** [comment]
Wonder why travis was green with this :/ https://travis-ci.org/sonata-project/SonataUserBundle/builds/90344289

Supply a quick fix the next hours.

**Comment by OskarStark** [comment]
indeed, travis was green...

**Comment by rande** [comment]
Don't trust test ;) coverage is not complete ...

**Comment by OskarStark** [comment]
ok, sorry

2015-11-18 14:26 GMT+01:00 Thomas <notifications@github.com>:

**Comment by rande** [comment]
No pb ;) It is always time to add them or ask people to contribute ;)

**Comment by Soullivaneuh** [comment]
> Don't trust test ;) coverage is not complete ...

It should be. :-P

**Pull request merged**